csharp/ql/src/utils/modelgenerator/internal/CaptureModelsSpecific.qll

```diff
@@ -130,6 +130,21 @@ class TargetApiSpecific extends Callable {
   predicate isRelevant() { relevant(this) }
 }
 
+/**
+ * A class of members that should be considered tainted from a model generation
+ * perspective.
+ */
+private class TaintRelevantMembers extends TaintTracking::TaintedMember {
+  TaintRelevantMembers() {
+    this.isEffectivelyPublic() and
+    this.fromSource() and
+    (this.(Property).getSetter().isEffectivelyPublic() or this instanceof CS::Field) and
+    exists(DataFlow::ParameterNode p | p.getEnclosingCallable() instanceof TargetApiSpecific |
+      this.getDeclaringType() = p.getType()
+    )
```

Comment:
This restriction just says that the member must be declared in the type of _some_ parameter, it doesn't restrict reads of this member to be from parameters; is that deliberate?

Reply:
Hmm, no, that is not deliberate.